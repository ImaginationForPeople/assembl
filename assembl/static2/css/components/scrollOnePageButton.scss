.scroll-one-page {
  display: inline-block;
  position: fixed;
  cursor: pointer;
  z-index: 99;
  left: 50%;
  margin-left: -34px;

  span {
    border-radius: 40px 40px 0 0;
    height: 40px;
    width: 80px;
    display: block;
    color: $white;
    text-align: center;

    .icon {
      padding-top: 12px;
      margin-bottom: 2px;

      &:before {
        font-size: 16px;
        margin: 0 0 0 8px;
      }
    }
  }
}
