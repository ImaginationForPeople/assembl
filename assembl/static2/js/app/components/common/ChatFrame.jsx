// @flow
import React from 'react';
import { Translate } from 'react-redux-i18n';
import { OverlayTrigger, Tooltip } from 'react-bootstrap';
import { connect } from 'react-redux';
import get from 'lodash/get';

type Props = {
  src: string
};

type State = {
  isOpen: boolean
};

const ChatFrameModal = ({ src }: { src: string }) => (
  <div className="chatframe-modal">
    <div className="chatframe-modal-header">
      <span className="chatframe-icon chatframe-modal-icon assembl-icon-robot" />
      <Translate value="chatframe.title" />
    </div>
    <iframe className="chatframe-modal-iframe" title="chatframe" src={src} />
  </div>
);

const ChatFrameButton = ({ isOpen, toggle }: { isOpen: boolean, toggle: () => void }) =>
  (isOpen ? (
    <div onClick={toggle} className="chatframe-icon chatframe-button assembl-icon-cancel" />
  ) : (
    <OverlayTrigger
      placement="left"
      overlay={(
        <Tooltip id="chatframe-tooltip">
          <Translate value="chatframe.tooltip" />
        </Tooltip>
      )}
    >
      <div onClick={toggle} className="chatframe-icon chatframe-button assembl-icon-robot" />
    </OverlayTrigger>
  ));

class DumbChatFrame extends React.Component<Props, State> {
  state = { isOpen: false };

  toggle = () => this.setState(({ isOpen }) => ({ isOpen: !isOpen }));

  render = () => {
    const { src } = this.props;
    if (!src) return null;
    const isOpen = this.state.isOpen;
    return (
      <div className={`chatframe ${isOpen ? 'open' : ''}`}>
        {isOpen && <ChatFrameModal src={src} />}
        <ChatFrameButton isOpen={isOpen} toggle={this.toggle} />
      </div>
    );
  };
}

const ChatFrame = connect(state => ({
  src: get(state, 'debate.debateData.chatframe.src')
}))(DumbChatFrame);

export default ChatFrame;