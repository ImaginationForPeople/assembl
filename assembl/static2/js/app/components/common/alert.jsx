// @flow
import React from 'react';
import { Alert } from 'react-bootstrap';

type Props = {
  isBase: boolean
};

type State = {
  showAlert?: boolean,
  topPosition?: boolean,
  alertMsg?: string,
  base: boolean,
  alertStyle?: string
};

class AssemblAlert extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      base: true
    };
  }

  componentWillReceiveProps(nextProps: Props) {
    if (nextProps.isBase) {
      this.setState({
        base: true
      });
    }
  }

  render() {
    const { showAlert, alertStyle, alertMsg, topPosition, base } = this.state;
    if (base) {
      return <Alert className="hideAlert" />;
    }
    return (
      <Alert style={topPosition ? { top: 0 } : { margin: 0 }} className={showAlert ? 'showAlert' : 'hideAlert'} bsStyle={alertStyle}>
        {alertMsg}
      </Alert>
    );
  }
}

export default AssemblAlert;