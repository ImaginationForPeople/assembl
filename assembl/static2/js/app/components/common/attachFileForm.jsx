// @flow
import React from 'react';
import { Button } from 'react-bootstrap';
import { Translate } from 'react-redux-i18n';

import FileUploader from './fileUploader';

type AttachFileFormProps = {
  onSubmit: Function
};

type AttachFileFormState = {
  file: File | null
};

class AttachFileForm extends React.Component<*, AttachFileFormProps, AttachFileFormState> {
  props: AttachFileFormProps;

  state: AttachFileFormState;

  fileInput: HTMLInputElement;

  constructor() {
    super();
    this.state = {
      file: null
    };
  }

  handleSubmit = (): void => {
    this.props.onSubmit(this.state.file);
  };

  handleFileChange = (file: File): void => {
    this.setState({
      file: file
    });
  };

  render() {
    let isImage;
    const file = this.state.file;
    if (file !== null) {
      isImage = file.type.includes('image');
    }

    return (
      <div className={isImage ? 'attach-file with-preview' : 'attach-file'}>
        <div htmlFor="attachment" className="attachment-label">
          <Translate value="common.attachFileForm.label" />
        </div>

        <FileUploader handleChange={this.handleFileChange} fileOrUrl={this.state.file} withPreview />
        <Button className="button-submit button-dark btn btn-default" onClick={this.handleSubmit}>
          <Translate value="common.attachFileForm.submit" />
        </Button>
      </div>
    );
  }
}

export default AttachFileForm;