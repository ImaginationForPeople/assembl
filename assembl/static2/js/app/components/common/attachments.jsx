// @flow
import React from 'react';
import { Translate } from 'react-redux-i18n';

export type Document = {
  id: string,
  title?: string,
  externalUrl: string,
  mimeType?: string,
  file?: File
};

export type Attachment = {
  entityKey: string,
  id?: string,
  document: Document
};

type AttachmentsProps = {
  attachments: Array<Attachment>
};

const DEFAULT_FILENAME = 'file';

const Attachments = ({ attachments }: AttachmentsProps) => (
  <div className="attachments">
    {attachments.map((attachment) => {
      const { externalUrl, mimeType, title } = attachment.document;
      if (mimeType && mimeType.startsWith('image/')) {
        return null;
      }

      return (
        <div className="attachment" key={attachment.id}>
          <span className="assembl-icon-synthesis" />
          <span className="title">{title || externalUrl}</span>
          <a download={title || DEFAULT_FILENAME} href={externalUrl} type={mimeType} target="_blank" rel="noopener noreferrer">
            <Translate value="common.attachments.download" />
          </a>
        </div>
      );
    })}
  </div>
);

export default Attachments;