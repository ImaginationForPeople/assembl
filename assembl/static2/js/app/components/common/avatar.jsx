// @flow
import React from 'react';
import { connect } from 'react-redux';
import { compose, graphql } from 'react-apollo';
import { Translate } from 'react-redux-i18n';
import { NavDropdown, MenuItem } from 'react-bootstrap';

import { getContextual, get } from '../../utils/routeMap';
import UserQuery from '../../graphql/userQuery.graphql';
import manageErrorAndLoading from './manageErrorAndLoading';
import { browserHistory } from '../../router';
import { localAwareLink } from '../../utils/utilityManager';

export type Props = {
  // used by getDerivedStateFromProps
  location: string, // eslint-disable-line
  slug: string,
  connectedUserId: string,
  loginData: {
    url: string,
    local: boolean
  },
  displayName: string,
  split: ?boolean
};

type State = { next: string };

const LoginAnchor = () => (
  <div className="connection">
    <Translate value="navbar.connection" />
  </div>
);

export const LocalAwareAnchor = localAwareLink(LoginAnchor);

class Avatar extends React.Component<Props, State> {
  static getDerivedStateFromProps(props: Props) {
    if (props.location) {
      return { next: props.location };
    }
    return null;
  }

  state = { next: '' };

  render() {
    const { slug, connectedUserId, loginData, displayName, split } = this.props;
    let loginUrl = `${getContextual('login', { slug: slug })}?next=${this.state.next}`;
    if (loginData && loginData.url) {
      loginUrl = loginData.url.includes('?') ? `${loginData.url}&next=${this.state.next}` : `${loginData.url}?next=${this.state.next}`;
    }
    const dropdownUser = (
      <div className="inline">
        <span className="assembl-icon-profil grey" />
        <span className="user-account">{displayName && displayName.length >= 17 && split ? `${displayName.substring(0, 17)}...` : displayName}</span>
      </div>
    );

    const urlData = { url: loginUrl, local: loginData.local };
    return (
      <div className="right avatar">
        {!connectedUserId && loginUrl && <LocalAwareAnchor urlData={urlData} />}
        {connectedUserId && (
          <ul className="dropdown-xs">
            <NavDropdown pullRight title={dropdownUser} id="user-dropdown">
              <MenuItem
                onClick={() => {
                  browserHistory.push(get('profile', { slug: slug, userId: connectedUserId }));
                }}
              >
                <Translate value="profile.panelTitle" />
              </MenuItem>
              <MenuItem id="logout-dropdown" href={`${getContextual('oldLogout', { slug: slug })}?next=${get('home', { slug: slug })}`}>
                <Translate value="navbar.logout" />
              </MenuItem>
            </NavDropdown>
          </ul>
        )}
      </div>
    );
  }
}

const mapStateToProps = ({ context, debate }) => ({
  slug: debate.debateData.slug,
  connectedUserId: context.connectedUserId,
  id: context.connectedUserIdBase64
});

export { Avatar };

export default compose(
  connect(mapStateToProps),
  graphql(UserQuery, {
    skip: props => !props.id,
    props: ({ data }) => {
      if (data.error || data.loading) {
        return {
          error: data.error,
          loading: data.loading
        };
      }

      return {
        error: data.error,
        loading: data.loading,
        displayName: data.user.displayName
      };
    }
  }),
  manageErrorAndLoading({ displayLoader: false })
)(Avatar);