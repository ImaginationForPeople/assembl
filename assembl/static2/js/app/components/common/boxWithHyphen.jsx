// @flow
import classnames from 'classnames';
import React from 'react';
import { Localize } from 'react-redux-i18n';

type BoxProps = {
  body: string,
  date: string, // TODO:  date iso format type checking
  hyphenStyle: Object,
  subject: string,
  title: string
};

type BoxWithHyphenProps = {
  additionalContainerClassNames: string,
  body: string,
  date: string, // TODO:  date iso format type checking
  href: string,
  hyphenStyle: Object,
  subject: string,
  title: string
};

const Box = ({ body, date, hyphenStyle, subject, title }: BoxProps) => (
  <div className="insert-box">
    <h3 className="dark-title-4 ellipsis">
      <div>{title}</div>
      <div className="ellipsis margin-xs">{subject}</div>
    </h3>
    <div className="box-hyphen" style={hyphenStyle}>
      &nbsp;
    </div>
    <div className="date">{date ? <Localize value={date} dateFormat="date.format2" /> : null}</div>
    <div className="insert-content margin-s">
      <p dangerouslySetInnerHTML={{ __html: body }} />
    </div>
  </div>
);

class BoxWithHyphen extends React.Component<*, BoxWithHyphenProps, *> {
  render() {
    const { additionalContainerClassNames, href, ...otherProps } = this.props;
    const containerClassNames = classnames([additionalContainerClassNames, 'box-with-hyphen']);
    return (
      <div className={containerClassNames}>
        {href.length > 1 ? (
          <a href={href}>
            <Box {...otherProps} />
          </a>
        ) : (
          <Box {...otherProps} />
        )}
      </div>
    );
  }
}

BoxWithHyphen.defaultProps = {
  additionalContainerClassNames: '',
  date: '',
  href: '',
  hyphenStyle: {}
};

export default BoxWithHyphen;