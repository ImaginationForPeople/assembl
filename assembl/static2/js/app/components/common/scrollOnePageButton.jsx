// @flow
import React from 'react';

import { isMobile } from '../../utils/globalFunctions';
import { withScreenHeight } from './screenDimensions';

type Props = {
  hidden: boolean,
  screenHeight: number
};

const scrollOnePageDown = (screenHeight: number) => () => {
  window.scrollTo({ top: screenHeight, left: 0, behavior: 'smooth' });
};

export const DumbScrollOnePageButton = ({ hidden, screenHeight }: Props) => {
  const scrollOnePageTopPosition = screenHeight - 35;
  const isTouchScreen = isMobile.any();
  return (
    <button
      type="button"
      className={`scroll-one-page no-border no-background ${hidden || screenHeight > 750 || isTouchScreen ? 'hidden' : ''}`}
      onClick={scrollOnePageDown(screenHeight)}
      style={{ top: scrollOnePageTopPosition }}
    >
      <span>
        <span className="icon assembl-icon-down-open">&nbsp;</span>
      </span>
    </button>
  );
};

export default withScreenHeight(DumbScrollOnePageButton);