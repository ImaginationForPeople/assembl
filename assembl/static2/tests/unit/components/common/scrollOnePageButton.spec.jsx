import React from 'react';
import renderer from 'react-test-renderer';

import { DumbScrollOnePageButton } from '../../../../js/app/components/common/scrollOnePageButton';
import '../../../helpers/setupTranslations';

describe('DumbScrollOnePageButton component', () => {
  it('should render a button to scroll one page down', () => {
    const component = renderer.create(<DumbScrollOnePageButton screenHeight={768} hidden={false} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a button to scroll one page down that is hidden', () => {
    const component = renderer.create(<DumbScrollOnePageButton screenHeight={768} hidden />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});