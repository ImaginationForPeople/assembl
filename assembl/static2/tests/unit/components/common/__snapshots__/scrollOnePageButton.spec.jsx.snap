// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DumbScrollOnePageButton component should render a button to scroll one page down 1`] = `
<button
  className="scroll-one-page no-border no-background hidden"
  onClick={[Function]}
  style={
    Object {
      "top": 733,
    }
  }
  type="button"
>
  <span>
    <span
      className="icon assembl-icon-down-open"
    >
       
    </span>
  </span>
</button>
`;

exports[`DumbScrollOnePageButton component should render a button to scroll one page down that is hidden 1`] = `
<button
  className="scroll-one-page no-border no-background hidden"
  onClick={[Function]}
  style={
    Object {
      "top": 733,
    }
  }
  type="button"
>
  <span>
    <span
      className="icon assembl-icon-down-open"
    >
       
    </span>
  </span>
</button>
`;
